### PR TITLE
Add Multimodal to generate() Java API

### PR DESCRIPTION
**Pull request by fionabos**
Updated the generate() Java API to support multimodal models for the Phi-3 Vision app.

- Added MultiModalProcessor.java, Images.java, NamedTensors.java classes and corresponding .jni files
- Updated GeneratorParams class to include setInputs(NamedTensors) method
- Added MultiModalProcessorTest.java

### CODE REVIEW

File: VERSION_INFO [hunk 1]
```diff
@@ -1 +1 @@
-0.4.0-dev
\ No newline at end of file
+0.4.1-dev
```
Comment (natke) [inline VERSION_INFO:1]: The version should not be updated by this PR

Comment (fionabos) [reply VERSION_INFO:1]: fixed

[the remark this answers is not in the document]
Comment (fionabos) [reply VERSION_INFO:1]: fixed

File: src/java/src/test/java/ai/onnxruntime/genai/MultiModalProcessorTest.java
[the remark this answers is not in the document]
Comment (fionabos) [reply src/java/src/test/java/ai/onnxruntime/genai/MultiModalProcessorTest.java:1]: I copied the formatting of Scott's tokenizer test. Nothing was checking in on the other tests, so unsure if/how to check in phi3 vision model if needed.

**Review by baijumeswani — APPROVED** [review]

**Pull request merged**